api/krusty/nameprefixsuffixpatch_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// I expected configMapRef.name and configMapKeyRef.name to be "mysql-g9ttm44c48" not "mysql"
+
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+func TestNamePrefixSuffixPatch(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PrefixSuffixTransformer").
+		PrepBuiltin("AnnotationsTransformer").
+		PrepBuiltin("LabelTransformer")
+	defer th.Reset()
+
+	th.WriteF("/app/handlers/kustomization.yaml", `
+nameSuffix: -suffix
+
+resources:
+  - ./deployment.yaml
+`)
+
+	th.WriteF("/app/handlers/deployment.yaml", `
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: short
+spec:
+  template:
+    spec:
+      containers:
+      - name: handler
+`)
+
+	th.WriteF("/app/mysql/kustomization.yaml", `
+configMapGenerator:
+  - name: mysql
+    literals:
+      - MYSQL_USER=mysql
+      - MYSQL_DATABASE=default
+`)
+
+	th.WriteK("/app", `
+resources:
+  - ./mysql
+  - ./handlers
+
+configMapGenerator:
+  - name: mysql
+    behavior: merge
```

Comment:
This replaces _all the fields_ in the config map, with no "savings" in terms of configuration,
so the layout doesn't make sense in this isolated context.

So let's merge more variables in here to make it realistic.

Without merging in more vars, the immediate fix is this layout:
```
func TestNamePrefixSuffixPatchAlternative(t *testing.T) {
	th := kusttest_test.MakeHarness(t)

	th.WriteF("handlers/kustomization.yaml", `
nameSuffix: -suffix
resources:
- deployment.yaml
`)

	th.WriteF("handlers/deployment.yaml", `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: short
spec:
  template:
    spec:
      containers:
      - name: handler
`)

	th.WriteK(".", `
resources:
- handlers
configMapGenerator:
- name: mysql
  literals:
  - MYSQL_DATABASE=db
  - MYSQL_USER=user
patches:
- target:
    kind: Deployment
    name: s.*
  patch: |-
    kind: Deployment
    metadata:
      name: ignored
    spec:
      template:
        spec:
          containers:
            - name: handler
              envFrom:
                - configMapRef:
                    name: mysql
              env:
                - valueFrom:
                    configMapKeyRef:
                      name: mysql
                      key: MYSQL_DATABASE
`)

	m := th.Run(".", th.MakeDefaultOptions())
	th.AssertActualEqualsExpected(m, `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: short-suffix
spec:
  template:
    spec:
      containers:
      - env:
        - valueFrom:
            configMapKeyRef:
              key: MYSQL_DATABASE
              name: mysql-g9ttm44c48
        envFrom:
        - configMapRef:
            name: mysql-g9ttm44c48
        name: handler
---
apiVersion: v1
data:
  MYSQL_DATABASE: db
  MYSQL_USER: user
kind: ConfigMap
metadata:
  name: mysql-g9ttm44c48
`)
}
```

Reply:
I made it a little more realistic.